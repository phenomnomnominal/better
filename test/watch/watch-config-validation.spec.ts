import { describe, expect, it } from 'vitest';

import { createFixture } from '../fixture.js';

describe('betterer.watch', () => {
  it.each([{ ignores: 1234 }, { ignores: true }, { ignores: {} }, { ignores: [1234] }])(
    `should throw when there is invalid config: %s`,
    async (config) => {
      const { betterer } = await import('@betterer/betterer');

      const { cleanup, logs, paths } = await createFixture('watch-config-validation', {
        '.betterer.js': ''
      });

      const configPaths = [paths.config];
      const resultsPath = paths.results;

      // @ts-expect-error testing invalid config
      await expect(async () => await betterer.watch({ configPaths, resultsPath, ...config })).rejects.toThrow();

      expect(logs).toMatchSnapshot();

      await cleanup();
    }
  );
});
