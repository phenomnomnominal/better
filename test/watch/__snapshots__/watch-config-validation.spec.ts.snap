// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`betterer.watch > should throw when there is invalid config: { ignores: [ 1234 ] } 1`] = `
[
  "
Error: "ignores" must be an array of strings. Received \`[1234]\`.
",
]
`;

exports[`betterer.watch > should throw when there is invalid config: { ignores: {} } 1`] = `
[
  "
Error: "ignores" must be an array of strings. Received \`[{}]\`.
",
]
`;

exports[`betterer.watch > should throw when there is invalid config: { ignores: 1234 } 1`] = `
[
  "
Error: "ignores" must be an array of strings. Received \`[1234]\`.
",
]
`;

exports[`betterer.watch > should throw when there is invalid config: { ignores: true } 1`] = `
[
  "
Error: "ignores" must be an array of strings. Received \`[true]\`.
",
]
`;
